src/main/java/com/iexec/common/chain/eip712/EIP712Utils.java

```diff
@@ -72,7 +72,7 @@ static String encodeBigInteger(BigInteger bigInteger) {
     }
 
     static String encodeByteArray(byte[] byteArray) {
-        if (!BytesUtils.isBytes32(byteArray)) {
+        if (!BytesUtils.isBytes32(BytesUtils.bytesToString(byteArray))) {
```

Comment:
`BytesUtils.bytesToString(byteArray)` calls `Numeric.toHexString(bytes)`, so it looks we are calling twice the method until the `return`.

I might say checking `bytes != null && bytes.length == BYTES_32_SIZE` in the `if ` would be more appropriate

Reply:
Sure, thanks!